sys/phydat/phydat_str.c

```diff
@@ -64,6 +64,8 @@ const char *phydat_unit_to_str(uint8_t unit)
         case UNIT_TEMP_K:   return "K";
         case UNIT_LUX:      return "lx";
         case UNIT_M:        return "m";
+        case UNIT_M2:       return "sqm";
+        case UNIT_M3:       return "cbm";
```

Comment:
Ah good idea, too - I came up with the following, but like yours too:

```
        case 3:     ((unit == UNIT_M2) || (unit == UNIT_M3)) ? return '\0' : return 'k';
        case 6:     (unit == UNIT_M2) ? return 'k' : return 'M';
        case 9:     (unit == UNIT_M3) ? return 'k' : return 'G';
```

Btw. the API change is only internal but necessary for any multi dimensional UNIT, i.e., areas and volumes.